graphs-lab/src/main/kotlin/graphs_lab/algs/DijkstraAlgorithm.kt

```diff
@@ -0,0 +1,170 @@
+	 * is not contained in the graph
+	 * @throws IllegalStateException if there is an appeal to a non-existent identifier
+	 */
+	fun getPath(idSource: I, idTarget: I): MutableList<I> {
+		val path = mutableListOf<I>()
+
+		if (statusTable == null || checkAndGetFirst(statusTable) != idSource) {
+			getAllPaths(idSource)
+		}
+		check(checkAndGetSecond(statusTable)) { "All paths do not build." }
+		require(table[idTarget] != null) { "The target vertex is not contained." }
+		if (table[idTarget]?.first == null) return path
+
+		// Assembles the path in the reversed order of movement
+		var idCurrent = idTarget
+		path.add(idCurrent)
+		while (idCurrent != idSource) {
+			idCurrent = checkAndGetFirst(table[idCurrent])
+			path.add(idCurrent)
+		}
+
+		return path.asReversed()
+	}
+
+	/**
+	 * Returns a weight of the shortest path from a source vertex to a target vertex.
+	 *
+	 * @param idSource the identifier of the source vertex
+	 * @param idTarget the identifier of the target vertex
+	 * @return a weight equals the sum of the all weights of edges on the path from the given identifier
+	 * @throws IllegalArgumentException if a weight is negative number or [idSource] or [idTarget]
+	 * is not contained in the graph
+	 * @throws IllegalStateException if there is an appeal to a non-existent identifier
+	 */
+	fun getWeightPath(idSource: I, idTarget: I): Double {
+		if (statusTable == null || checkAndGetFirst(statusTable) != idSource) {
+			getAllPaths(idSource)
+		}
+		check(checkAndGetSecond(statusTable)) { "All paths do not build." }
+		require(table[idTarget] != null) { "The target vertex is not contained." }
+		if (table[idTarget]?.first == null) return Double.POSITIVE_INFINITY
+
+		return checkAndGetSecond(table[idTarget])
+	}
+
+	/**
+	 * @param idSource the identifier of the source vertex
+	 * @return [table]
+	 * @throws IllegalArgumentException if a weight is negative number or [idSource] is not contained in the graph
+	 * @throws IllegalStateException if there is an appeal to a non-existent identifier
+	 */
+	fun getAllPaths(idSource: I): MutableMap<I, Pair<I?, Double>> {
+		require(idSource in graph.idVertices) { "The source vertex is not contained." }
+		statusTable = Pair(idSource, false)
+
+		val statusVertices = mutableMapOf<I, Boolean>()
+
+		// Initializes `table` and `statusVertices` on the first step
+		graph.idVertices.forEach {
+			table[it] = Pair(null, Double.POSITIVE_INFINITY)
+			statusVertices[it] = false
+		}
+		table[idSource] = Pair(null, 0.0)
+		statusVertices[idSource] = true
+		var countVisited = 1
+
+		if (graph.vertexEdges(idSource).isEmpty()) {
+			statusTable = Pair(idSource, true)
+			return table
+		}
+
+		val heapEdges = mutableListOf<WeightedEdge<I>>()
+		var edgeNext: WeightedEdge<I>? = null
+
+		// Searches the edge with the minimal weight from the source vertex
+		graph.vertexEdges(idSource).forEach {
+			require(it.weight > 0) { "Negative number does not support." }
```

Comment:
А почему мы не поддерживаем ребра, у которых `weight == 0`. Вроде как алгоритм Дейкстры может некорректно работать с отрицательными весами, а ноль неотрицателен, или ты стремишься к 0-. По идее тут должно быть так:
```suggestion
			require(it.weight >= 0) { "Negative number does not support." }
```